Interpolater.C

```diff
@@ -0,0 +1,181 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2021, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Description: Implementation of the Interpolater algorithm.
+
+#include "Interpolater.h"
+
+#include <limits.h>
+#include <cmath>
+#include "Matrix.h"
+#include "scalapack_wrapper.h"
+#include "mpi.h"
+
+/* Use C++11 built-in shared pointers if available; else fallback to Boost. */
+#if __cplusplus >= 201103L
+#include <memory>
+#else
+#include <boost/shared_ptr.hpp>
+#endif
+
+using namespace std;
+
+namespace CAROM {
+
+Interpolater::Interpolater(std::vector<Vector*> parameter_points,
+                           std::vector<Matrix*> rotation_matrices,
+                           int ref_point,
+                           double epsilon)
+{
+    CAROM_VERIFY(parameter_points.size() == rotation_matrices.size());
+    CAROM_VERIFY(parameter_points.size() > 0);
+
+    // Get the rank of this process, and the number of processors.
+    int mpi_init;
+    MPI_Initialized(&mpi_init);
+    if (mpi_init == 0) {
+        MPI_Init(nullptr, nullptr);
+    }
+
+    MPI_Comm_rank(MPI_COMM_WORLD, &d_rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &d_num_procs);
+
+    d_parameter_points = parameter_points;
+    d_rotation_matrices = rotation_matrices;
+    d_ref_point = ref_point;
+    d_lambda_T = NULL;
+    d_epsilon = epsilon;
+}
+
+std::vector<double> Interpolater::obtainRBF(Vector* point)
+{
+    // Inverse quadratic RBF
+    std::vector<double> inv_q;
+    for (int i = 0; i < d_parameter_points.size(); i++)
+    {
+        Vector diff;
+        point->minus(*d_parameter_points[i], diff);
+        double res = 1.0 / (1.0 + std::pow(d_epsilon*diff.norm(), 2));
```

Comment:
I'm not sure if this function is a bottleneck, but `x*x` may be significantly faster than `pow(x, 2)`.

Reply:
Even better would be to implement a `norm2` function for the norm squared, which avoids the even more costly `sqrt` call and the `pow` call.